core/server/src/main/java/alluxio/master/AlluxioMaster.java

```diff
@@ -154,7 +42,7 @@ public static void main(String[] args) {
   /**
    * @return the (cached) master service loader
    */
-  private static ServiceLoader<MasterFactory> getServiceLoader() {
+  protected static ServiceLoader<MasterFactory> getServiceLoader() {
```

Comment:
I feel this and the next static method should be a part of  interface `AlluxioMasterService.Factory` (See my next comment), and leave this class really a simple wrapper